### PR TITLE
test_linearizer cleanup

### PR DESCRIPTION
**Pull request by Qazalin**
cleans up the skips for running linearizer in compiled backends only

### CODE REVIEW

File: test/test_linearizer.py [hunk 1]
```diff
@@ -13,10 +14,14 @@
...
 from tinygrad.helpers import dtypes, prod
 
-class TestLinearizer(unittest.TestCase):
+class _BaseTestLinearizer(unittest.TestCase):
+  device: Compiled = cast(Compiled, Device[Device.DEFAULT])
+  @classmethod
+  def setUpClass(cls):
+    if not isinstance(Device[Device.DEFAULT], Compiled): raise unittest.SkipTest("Only Compiled uses linearizer")
+
+class TestLinearizer(_BaseTestLinearizer):
```
Comment (chenyuxyz) [inline test/test_linearizer.py:23]: why not just decorate the class with `@unittest.skipIf(...)`?

Comment (Qazalin) [reply test/test_linearizer.py:23]: every single one of them? I guess you could yeah but setup() is also pretty easy to just inherit

Comment (chenyuxyz) [reply test/test_linearizer.py:23]: yeah that's what i would do. it's more local so a lot easier to read and move around.

Comment (Qazalin) [reply test/test_linearizer.py:23]: does it really make sense for any linearizer test to run on non-compiled backends? that's why I kinda wanted it to be a blackbox

Comment (chenyuxyz) [reply test/test_linearizer.py:23]: no we skip non compiled backend on linearizer tests. there are other files that runs linearizer tests too. do they need to import and inherit from this file? i don't see a good reason to add this complexity or make it a blackbox

**Comment by chenyuxyz** [comment]
cool thanks!

**Pull request merged**